ffi/pkg/shadow/ffi/mod.go

```diff
@@ -55,20 +55,20 @@ func EpochWait(blockno uint64) bool {
 }
 
 //export Proof
-func Proof(api string, number uint64) *C.char {
+func Proof(api string, number uint64, block bool) (*C.char, *C.char) {
     header, err := eth.Header(api, number)
     if err != nil {
         log.Error("get ethashproof when get header failed %v", err)
-        return C.CString("")
+        return nil, C.CString(err.Error())
     }
 
-    _, proof, err := ethproof.Proof(&header, true)
+    _, proof, err := ethproof.Proof(&header, block)
```

Comment:
Suggest to rename `block` to `wait` to avoid ambiguity with blockchain's `block`